### ISSUE TITLE
Does not seem to support React forwardRef

### ISSUE THREAD

**Issue by ericmatte** [airbnb/ts-migrate#9]
Hi,

This project is great! However, it doesn't seem to support well forwardRefs in React.
I'm not sure if this is an issue or a feature request, but here is an example with a simple `input`:

### Without `forwardRef`

<details>
<summary>JS file</summary>
<br>

```jsx
import React, { memo } from "react";
import PropTypes from "prop-types";

const propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func,
};

const defaultProps = {
  className: "",
  onChange: undefined,
};

const TestInput = ({ label, className, onChange, ...props }) => {
  return (
    <div className={className}>
      <label>
        {label}
        <input onChange={onChange} {...props} />
      </label>
    </div>
  );
};

TestInput.propTypes = propTypes;
TestInput.defaultProps = defaultProps;

export default memo(TestInput);
```

</details>

<details>
<summary>TS file after migration</summary>
<br>

```tsx
import React, { memo } from "react";

type OwnProps = {
  label: string;
  className?: string;
  onChange?: (...args: any[]) => any;
};

const defaultProps = {
  className: "",
  onChange: undefined,
};

type Props = OwnProps & typeof defaultProps;

const TestInput = ({ label, className, onChange, ...props }: Props) => {
  return (
    <div className={className}>
      <label>
        {label}
        <input onChange={onChange} {...props} />
      </label>
    </div>
  );
};
TestInput.defaultProps = defaultProps;

export default memo(TestInput);

```

</details>

---

### With `forwardRef`

<details>
<summary>JS file</summary>
<br>

```jsx
import React, { memo, forwardRef } from "react";
import PropTypes from "prop-types";

const propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func,
};

const defaultProps = {
  className: "",
  onChange: undefined,
};

const TestInput = forwardRef(({ label, className, onChange, ...props }, ref) => {
  return (
    <div className={className}>
      <label>
        {label}
        <input ref={ref} onChange={onChange} {...props} />
      </label>
    </div>
  );
});

TestInput.propTypes = propTypes;
TestInput.defaultProps = defaultProps;

export default memo(TestInput);
```

</details>

<details>
<summary><b>Actual</b> TS file after migration</summary>
<br>

PropTypes are not converted to types.

```tsx
import React, { memo, forwardRef } from "react";
import PropTypes from "prop-types";

const propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func,
};

const defaultProps = {
  className: "",
  onChange: undefined,
};

// @ts-expect-error ts-migrate(2339) FIXME: Property 'label' does not exist on type '{ childre... Remove this comment to see the full error message
const TestInput = forwardRef(({ label, className, onChange, ...props }, ref) => {
  return (
    <div className={className}>
      <label>
        {label}
        {/* @ts-expect-error ts-migrate(2322) FIXME: Type 'unknown' is not assignable to type 'HTMLInpu... Remove this comment to see the full error message */}
        <input ref={ref} onChange={onChange} {...props} />
      </label>
    </div>
  );
});

// @ts-expect-error ts-migrate(2559) FIXME: Type '{ label: Validator<string>; className: Requi... Remove this comment to see the full error message
TestInput.propTypes = propTypes;
// @ts-expect-error ts-migrate(2559) FIXME: Type '{ className: string; onChange: undefined; }'... Remove this comment to see the full error message
TestInput.defaultProps = defaultProps;

export default memo(TestInput);

```

</details>

<details>
<summary><b>Expected</b> TS file after migration</summary>
<br>

```tsx
import React, { memo, forwardRef } from "react";

type OwnProps = {
  label: string;
  className?: string;
  onChange?: (...args: any[]) => any;
};

const defaultProps = {
  className: "",
  onChange: undefined,
};

type Props = OwnProps & typeof defaultProps;

const TestInput = forwardRef<HTMLInputElement, Props>(({ label, className, onChange, ...props }, ref) => {
  return (
    <div className={className}>
      <label>
        {label}
        <input ref={ref} onChange={onChange} {...props} />
      </label>
    </div>
  );
});
TestInput.defaultProps = defaultProps;

export default memo(TestInput);

```

</details>

Regards

**Comment by Rudeg** [comment]
Yes, we didn't support it. 
It looks like an enhancement for the react-props plugin.

**Comment by lencioni** [comment]
Fixed by #23
